lib/nm/index.ts

```diff
@@ -0,0 +1,405 @@
+			];
+			await this.requestScan(requestScanParams);
+			return this.getAccessPoints(this.devices.wifi.path);
+		} catch (err) {
+			if (_.isArray(err) && this.ignoreScanStatus(err[0])) {
+				return this.getAccessPoints(this.devices.wifi.path);
+			}
+			throw formatError(500, `Could not listNearbyNetworks`, err);
+		}
+	}
+
+	ignoreScanStatus = (status) => {
+		return status === 'Scanning not allowed while already scanning' ||
+			status === 'Scanning not allowed immediately following previous scan' ||
+			status === 'Scanning not allowed while unavailable or activating';
+	}
+
+	getAccessPoints = async (wifiDevicePath) => {
+		const {AccessPoints} = await this.getObjectProperty(['org.freedesktop.NetworkManager.Device.Wireless', 'AccessPoints'], wifiDevicePath);
+		const getApsProperties = () => _.map(AccessPoints, this.getApProperties);
+		const rawAccessPoints = await Bluebird.all(getApsProperties());
+		const accessPoints = _.map(rawAccessPoints, (rawProps: any[]) => {
+			const props = _.reduce(rawProps, (acc, prop: any) => {
+				return {
+					...acc,
+					...prop,
+				};
+			}, {});
+			return props;
+		});
+		this.accessPoints = accessPoints;
+		return makeNetworksReadable(this.accessPoints);
+	}
+
+	getApProperties = async (path) => {
+		try {
+			const props = await Bluebird.all([
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Ssid'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Flags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'WpaFlags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'RsnFlags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Frequency'], path),
+			]);
+			return props;
+		} catch (err) {
+			throw formatError(500, `Could not get ap ${path} property`, err);
+		}
+	}
+
+	getCurrentNetwork = async () => {
+		try {
+			const [key, [connections]] = await this.getActiveConnections();
+			const getConnectionsType = () => _.map(connections, _.partial(this.getObjectProperty, ['org.freedesktop.NetworkManager.Connection.Active', 'Type']));
+			const results = await Bluebird.all(getConnectionsType());
+			const wifiConnection: any = _.filter(results, ({Type}) => Type === '802-11-wireless')[0];
+			if (_.isUndefined(wifiConnection)) {
+				return;
+			}
+			const {Id} = await this.getObjectProperty(['org.freedesktop.NetworkManager.Connection.Active', 'Id'], wifiConnection.path);
+			return Id;
+		} catch (err) {
+			if (err === 404) {
+				throw formatError(404, 'You\'re not currently connected to a wireless network');
+			}
+			throw formatError(500, `Could not getCurrentNetwork`, err);
+		}
+	}
+}
+
+function formatError(code: number = 400, message: string, err: any = {}) {
+	if (err.code) {
+		return err;
+	}
+	const error = new NetworkManagerError(message);
+	error.data = err;
+	error.code = String(code);
+	console.error(error);
```

Comment:
Nope, I did that on purpose to have the error also in the container's logs, otherwise it's only in the app container using the module.
Should I delete this?

Reply:
Library modules should not be logging errors, that's up to the caller to do. Where, or which container this is running in should be irrelevant to it's error handling, and the application code using this in the container will have to do error handling (and logging if applicable).